### ISSUE TITLE
useSentry: Option to exclude errors from reporting

### ISSUE THREAD

**Issue by dburles** [graphql-hive/envelop#925]
## Feature request

There's no way to exclude errors from being reported. Ideally there could be a configuration filter option to opt-out.

Something like:
```js
useSentry({
  skipError(error) { return error instanceof MyCustomError; }
});
```

I'd suggest renaming `skip` to `skipOperation` if we go with `skipError`.

## Bundling bug

I tried to extend my custom errors from [EnvelopError](https://github.com/dotansimha/envelop/blob/dc2ec27dc0ca783f37bcff2a8bba07dd94b8fb7f/packages/core/src/plugins/use-masked-errors.ts#L6-L10) as a workaround, though I was stumped as to why this didn't work until I realised for some reason it's actually being bundled into the Sentry plugin, https://unpkg.com/browse/@envelop/sentry@1.3.0/index.mjs so fails the [instanceof](https://github.com/dotansimha/envelop/blob/main/packages/plugins/sentry/src/index.ts#L167) check, when imported from `@envelop/core`.

**Comment by saihaj** [comment]
If you throw `EnvelopError` they are excluded from error reporting https://github.com/dotansimha/envelop/pull/855

**Comment by dburles** [comment]
That's what I was doing essentially, but it fails the [instanceof](https://github.com/dotansimha/envelop/blob/main/packages/plugins/sentry/src/index.ts#L167) check as it's [not the same class](https://unpkg.com/browse/@envelop/sentry@1.3.0/index.mjs) as in core.

**Comment by n1ru4l** [comment]
@dburles Can you create a minimal reproduction of this?

**Comment by dburles** [comment]
This issue has kind of become two, a bug report for the bundled `EnvelopError` class and a feature request for a configuration option to skip errors, the latter is the part I am interested in. I would love to help and create a repro for the `instanceof` issue, but I cannot devote the time to it at the minute. I believe the problem is clear in any case and that it is solved by not bundling mono repo dependencies.

**Comment by n1ru4l** [comment]
@dburles This will be taken care of by https://github.com/dotansimha/envelop/pull/936. I will let you know once it is merged and released!

**Comment by dotansimha** [comment]
> @dburles This will be taken care of by #936. I will let you know once it is merged and released!

@dburles anything needs to be done on this PR? I think the changes Laurin mentioned are already in.

**Comment by dotansimha** [comment]
Available in `@envelop/sentry@1.5.0` :)
